lib/arelastic/nodes/node.rb

```diff
@@ -7,7 +7,9 @@ class Node
 
       def convert_to_elastic(expr)
         if expr.is_a?(Array)
-          expr.map { |e| convert_to_elastic(e) }
+          expr = merge_expression(expr, klass: Queries::HasParent, merge_test_method: :parent_type)
+          expr = merge_expression(expr, klass: Queries::HasChild, merge_test_method: :type)
+          expr.uniq.map { |e| convert_to_elastic(e) }
```

Comment:
What is the ramification of duplicate expressions?

Reply:
When I see 5 MatchAll's, it makes it harder to visualize what the rest of the terms are, so debugging requires more mental overhead than necessary.  In terms of performance, I would hope that ES itself optimizes the query before running it so that there's no difference between 1 MatchAll and 5 of them, but I don't know.